packages/google-auth/src/googleAuth.ts

```diff
@@ -0,0 +1,57 @@
+import type { MiddlewareHandler } from 'hono'
+
+import { HTTPException } from 'hono/http-exception'
+import { AuthFlow } from './authFlow'
+
+export function googleAuth(options: {
+  scope: string[]
+  response_type: 'code' | 'token'
+  include_granted_scopes?: boolean
+  login_hint?: string
+  prompt?: 'none' | 'consent' | 'select_account'
+  state?: string
+  client_id?: string
+  client_secret?: string
+}): MiddlewareHandler {
+  return async (c, next) => {
+    // Create new Auth instance
+    const auth = new AuthFlow({
+      client_id: options.client_id || c.env?.GOOGLE_ID as string,
+      client_secret: options.client_secret || c.env?.GOOGLE_SECRET as string,
+      redirect_uri: c.req.url.split('?')[0],
+      response_type: options.response_type,
+      login_hint: options.login_hint,
+      prompt: options.prompt,
+      scope: options.scope,
+      include_granted_scopes: options.include_granted_scopes,
+      state: options.state,
+      code: c.req.query('code'),
+      token: {
+        token: c.req.query('access_token') as string,
+        expires_in: Number(c.req.query('expires-in')) as number
+      }
+    })
+
+    // Avoid CSRF attack by checking state
+    if (c.req.url.includes('?') && c.req.query('state') !== options.state) {
+      throw new HTTPException(401, { message: 'Unauthorized, risk of CSRF attack.' })
```

Comment:
I'd just return a 400/401 and remove the error message (you usually want to make these vague)

Reply:
Right, thanks!